gobblin-core/src/main/java/org/apache/gobblin/writer/CloseOnFlushWriterWrapper.java

```diff
@@ -104,13 +107,15 @@ public void close() throws IOException {
 
   @Override
   public void commit() throws IOException {
-    writer.commit();
+    if (!this.committed) {
```

Comment:
Is it expected behavior or unexpected behavior for commit() to be called multiple times? 
If unexpected, you should at least log a message (warn?) that you're skipping the commit.

Reply:
It can happen - if we end up with back-to-back control messages which is expected. (e.g. MetaDataUpdateControlMessage followed by FlushControlMessage). Currently, each control message will cause the writer to invoke commit(). But regardless, adding a log message is a good idea.